students/terrance_jones/lesson7/html_render.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3
+
+"""
+A class-based system for rendering html.
+"""
+
+
+# This is the framework for the base class
+class Element(object):
+    tag = "html"
+    indent = " "
+
+
+    def __init__(self, content=None, **kwargs):
+        if content is None:
+            self.contents = []
+        else:
+            self.contents = [content]
+        self.kwargs = kwargs
+        
+    def append(self, new_content):
+        self.contents.append(new_content)
+
+
+    def render(self, out_file, cur_ind=""):
+        #loop through the list of contents:
+        open_tag = ["<{}".format(self.tag)]
+        for key, value in self.kwargs.items():
+            open_tag.append(' {}="{}"'.format(key,value))
+        open_tag.append(">\n")
+        out_file.write("".join(open_tag))
+
+        for content in self.contents:
+            try: 
+                content.render(out_file)
+            except AttributeError:
+                out_file.write(str(content))
+                out_file.write("\n")
+        out_file.write("</{}>\n".format(self.tag))
+
+class Html(Element):
+    tag = 'html'
```

Comment:
Step 8:
> Update the Html element class to render the " < !DOCTYPE html > " tag at the head of the page, before the html element.
You can do this by subclassing Element, overriding render(), but then calling the Element render from the new render.

Per above directions, you should define a `render` method in this class and it would look something like:
```
def render(self, file_out, cur_ind=""):
        file_out.write(cur_ind + "<!DOCTYPE html>\n")
        super().render(file_out, cur_ind=cur_ind)
```